### PR TITLE
cmake: Add an option for enabling rook client in dashboard

### PR DESCRIPTION
**Pull request by wjwithagen**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -652,9 +652,9 @@ endif()
 set(DASHBOARD_FRONTEND_LANGS "" CACHE STRING
   "List of comma separated ceph-dashboard frontend languages to build. \
   Use value `ALL` to build all languages")
-
-# TODO: make this an option and set it to the same value as WITH_MGR_DASHBOARD_FRONTEND
-set(WITH_MGR_ROOK_CLIENT WITH_MGR_DASHBOARD_FRONTEND)
+if(WITH_MGR_DASHBOARD_FRONTEND)
+  option(WITH_MGR_ROOK_CLIENT "Include Rook in the DASHBOARD" ON)
+endif()
```
Comment (tchaikov) [inline CMakeLists.txt:657]:
```suggestion
CMAKE_DEPENDENT_OPTION(WITH_MGR_ROOK_CLIENT "Enable Rook support in the dashboard" ON
  "WITH_MGR_DASHBOARD_FRONTEND" OFF)
```

**Review by tchaikov — CHANGES_REQUESTED** [review]
could you please s/dashboard/cmake/ in the title of the commit message? and change the title to something like

> add an option for enabling rook client in dashboard

File: CMakeLists.txt (hunk 1 above)
Comment (wjwithagen) [reply CMakeLists.txt:657]: Always something new to learn....

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2021-04-15_08:31:03-rados-wip-kefu-testing-2021-04-15-1359-distro-basic-smithi/

**Comment by liewegas** [comment]
i think this is related?  testing a revert.
```
2021-04-16T14:07:14.885+0000 7f4640f77d00  1 mgr[py] Loading python module 'rook'
2021-04-16T14:07:14.898+0000 7f4640f77d00 10 mgr[py] Computed sys.path '/usr/share/ceph/mgr:/usr/local/lib64/python3.6/site-packages:/usr/local/lib/python3.6/site-packages:/usr/lib64/python3.6/site-packages:/usr/lib/python3.6/site-packages:/usr/lib64/python36.zip:/usr/lib64/python3.6:/usr/lib64/python3.6:/usr/lib64/python3.6/lib-dynload'
2021-04-16T14:07:15.103+0000 7f4633c1c700 10 cephx: validate_tickets want 55 have 55 need 0
2021-04-16T14:07:15.103+0000 7f4633c1c700 20 cephx client: need_tickets: want=55 have=55 need=0
2021-04-16T14:07:15.103+0000 7f4633c1c700 10 cephx client: build_rotating_request
2021-04-16T14:07:15.103+0000 7f4633c1c700  1 -- 172.21.15.105:0/30198 --> [v2:172.21.15.105:3301/0,v1:172.21.15.105:6790/0] -- auth(proto 2 2 bytes epoch 0) v1 -- 0x561ef2449b00 con 0x561ef2503400
2021-04-16T14:07:15.325+0000 7f4640f77d00 -1 mgr[py] Module not found: 'rook'
2021-04-16T14:07:15.325+0000 7f4640f77d00 -1 mgr[py] Traceback (most recent call last):
  File "/usr/share/ceph/mgr/rook/__init__.py", line 2, in <module>
    from .module import RookOrchestrator
  File "/usr/share/ceph/mgr/rook/module.py", line 38, in <module>
    from .rook_cluster import RookCluster
  File "/usr/share/ceph/mgr/rook/rook_cluster.py", line 38, in <module>
    from .rook_client.ceph import cephfilesystem as cfs
ModuleNotFoundError: No module named 'rook.rook_client.ceph'
```
/a/sage-2021-04-16_13:49:02-rados-wip-sage-testing-2021-04-16-0822-distro-basic-smithi/6051743

**Comment by wjwithagen** [comment]
@liewegas 
That was a test in theutology??

**Comment by liewegas** [comment]
Yep.

With revert: https://pulpito.ceph.com/sage-2021-04-16_19:04:25-rados-wip-sage4-testing-2021-04-16-1139-distro-basic-smithi/
Master: https://pulpito.ceph.com/sage-2021-04-18_12:58:03-rados-master-distro-basic-smithi/

**Comment by tchaikov** [comment]
should be fixed by #40901

**Pull request merged**